tests/remount_test.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volume
+
+import (
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = Describe("[csi] [jiva] TEST Remount when volume goes into RO mode", func() {
+	BeforeEach(prepareForRemountTest)
+	AfterEach(cleanupAfterRemountTest)
+
+	Context("App is deployed and restarted on pvc with replica count 1", func() {
+		It("Should run Volume Creation Test", remountTest)
+	})
+})
+
+func remountTest() {
+	By("creating and verifying PVC bound status", createAndVerifyPVC)
+	By("Creating and deploying app pod", createDeployVerifyApp)
+	By("scale down jiva controller pod", scaleDownControllerPod)
+	By("Verify app state as CrashLoopBackOff", func() { verifyCrashLoopBackOffStateOfAppPod(true) })
+	By("Scale up jiva controller pod", scaleUpControllerPod)
+	By("Verify app state as not CrashLoopBackOff", func() { verifyCrashLoopBackOffStateOfAppPod(false) })
```

Comment:
```suggestion
	By("Verify app state as Running", func() { verifyCrashLoopBackOffStateOfAppPod(false) })
```

Reply:
Actually if you look at the pod yaml, the status.phase feild always remains as Running even though the pod is in CrashLoopBackOff. That is the reason I used the above log.